devito/tools/data_structures.py

```diff
@@ -275,36 +281,62 @@ class PartialOrderTuple(tuple):
     items : object or iterable of objects
         The elements of the tuple.
     relations : iterable of tuples, optional
-        One or more binary relations between elements in ``items``. If not
-        provided, then ``items`` is interpreted as a totally ordered sequence.
+        One or more n-ary relations between the elements in `items`. If not
+        provided, then `items` is interpreted as a totally ordered sequence.
         If provided, then a (partial) ordering is computed and all elements in
-        ``items`` for which a relation is not provided are appended.
+        `items` for which a relation is not provided are appended.
+    mode : str, optional
+        If 'total' (default), the resulting object is interpreted as a totally
+        ordered sequence; the object's relations are simplified away and no
+        subsequent operation involving the Ordering will ever be able to alter
+        the obtained sequence. If 'partial', the outcome is a partially ordered
+        sequence; the relations as provided by the user are preserved, which
+        leaves room for further reordering upon future operations. If 'unordered',
+        the `relations` are ignored and the resulting object degenerates to an
+        unordered collection.
     """
-    def __new__(cls, items=None, relations=None):
+    def __new__(cls, items=None, relations=None, mode='total'):
+        assert mode in ('total', 'partial', 'unordered')
+
         items = as_tuple(items)
         if relations:
             items = cls.reorder(items, relations)
-        obj = super(PartialOrderTuple, cls).__new__(cls, items)
-        obj._relations = set(tuple(i) for i in as_tuple(relations))
+
+        obj = super().__new__(cls, items)
+
+        obj._relations = frozenset(cls.simplify_relations(relations, items, mode))
```

Comment:
to do what I think you have in mind, that is constructing the minimal set of relations such that you get the same (partial) ordering in output, requires a potentially expensive, non-trivial algorithm

but it's not what we do here

Some collections may be unordered, some may be partially ordered, and some totally ordered. For the totally ordered ones, we can indeed simplify away all of the relations and replace them with a single relation which is given essentially by the ordering itself

That's what I've used `simplify_relations` for, so far. It might be enhanced maybe